include/sdf/ParticleEmitter.hh

```diff
@@ -0,0 +1,352 @@
+    /// \return The pose of the particle emitter.
+    public: const ignition::math::Pose3d &RawPose() const;
+
+    /// \brief Set the pose of the particle emitter object.
+    /// \sa const ignition::math::Pose3d &RawPose() const
+    /// \param[in] _pose The pose of the particle emitter.
+    public: void SetRawPose(const ignition::math::Pose3d &_pose);
+
+    /// \brief Get the name of the coordinate frame relative to which this
+    /// emitter's pose is expressed. An empty value indicates that the frame is
+    /// relative to the parent link.
+    /// \return The name of the pose relative-to frame.
+    public: const std::string &PoseRelativeTo() const;
+
+    /// \brief Set the name of the coordinate frame relative to which this
+    /// emitter's pose is expressed. An empty value indicates that the frame is
+    /// relative to the parent link.
+    /// \param[in] _frame The name of the pose relative-to frame.
+    public: void SetPoseRelativeTo(const std::string &_frame);
+
+    /// \brief Get SemanticPose object of this object to aid in resolving poses.
+    /// \return SemanticPose object for this emitter.
+    public: sdf::SemanticPose SemanticPose() const;
+
+    /// \brief Get a pointer to the SDF element that was used during load.
+    /// \return SDF element pointer. The value will be nullptr if Load has
+    /// not been called.
+    public: sdf::ElementPtr Element() const;
+
+    /// \brief Get a pointer to the emitter's material properties. This can
+    /// be a nullptr if material properties have not been set.
+    /// \return Pointer to the emitter's material properties. Nullptr
+    /// indicates that material properties have not been set.
+    public: sdf::Material *Material() const;
+
+    /// \brief Set the emitter's material
+    /// \param[in] _material The material of the particle emitter.
+    public: void SetMaterial(const sdf::Material &_material);
+
+    /// \brief The path to the file where this element was loaded from.
+    /// \return Full path to the file on disk.
+    public: const std::string &FilePath() const;
+
+    /// \brief Set the path to the file where this element was loaded from.
+    /// \paramp[in] _filePath Full path to the file on disk.
+    public: void SetFilePath(const std::string &_filePath);
+
+    /// \brief Set the name of the xml parent of this object, to be used
+    /// for resolving poses. This is private and is intended to be called by
+    /// Link::SetPoseRelativeToGraph.
+    /// \param[in] _xmlParentName Name of xml parent object.
+    private: void SetXmlParentName(const std::string &_xmlParentName);
+
+    /// \brief Set a weak pointer to the PoseRelativeToGraph to be used
+    /// for resolving poses. This is private and is intended to be called by
+    /// Link::SetPoseRelativeToGraph.
+    /// \param[in] _graph Weak pointer to PoseRelativeToGraph.
+    private: void SetPoseRelativeToGraph(
+        std::weak_ptr<const PoseRelativeToGraph> _graph);
```

Comment:
`std::weak_ptr<const PoseRelativeToGraph>` has been replaced by `sdf::ScopedGraph<PoseRelativeToGraph>`

Reply:
Updated in 300e52f655d536661797edfe801dbb41cb67b3b8